addons/website_sale/controllers/main.py

```diff
@@ -392,6 +383,17 @@ def shop(self, page=0, category=None, search='', min_price=0.0, max_price=0.0, p
         fiscal_position_sudo = website.fiscal_position_id.sudo()
         products_prices = lazy(lambda: products._get_sales_prices(pricelist, fiscal_position_sudo))
 
+        attributes_values = request.env['product.attribute.value'].browse(attrib_set)
+        sorted_attributes_values = attributes_values.sorted('sequence')
+        multi_attributes_values = sorted_attributes_values.filtered(lambda av: av.display_type == 'multi')
+        single_attributes_values = sorted_attributes_values - multi_attributes_values
+        grouped_attributes_values = list(groupby(single_attributes_values, lambda av: av.attribute_id.id))
+        grouped_attributes_values.extend([(av.attribute_id.id, [av]) for av in multi_attributes_values])
+
+        selected_attributes = grouped_attributes_values and "#attribute_values=%s" % (
```

Comment:
`selected_attributes_hash` maybe ? (or something like that) would be a clearer variable name (and template rendering value too)